service/src/vsids.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::HashMap;
+use std::env;
+
+type Edge = (String, String, String);
+
+#[derive(Clone, Default)]
+pub struct VSIDSManager {
+    weights: HashMap<Edge, f64>,
```

Comment:
Actually, the implementation could be much more simple: for each `src` just store the single biggest "index" (vsids multiplier), and when that one crosses the normalization threshold, just renormalize every edge pointing out from `src`, memorize the biggest index and start subtracting it from every new incoming index.

Also, on every incoming edge, every outgoing edge for `src` must be checked if it is "infinitely smaller" than the new one, (e.g. maybe 10^5 smaller), and if so, _delete_ every edge that is too small, to free up resources.